x-pack/libbeat/management/managerV2.go

```diff
@@ -386,21 +386,28 @@ func (cm *BeatV2Manager) updateStatuses() {
 // Unit manager
 // ================================
 
-func (cm *BeatV2Manager) addUnit(unit *client.Unit) {
+func (cm *BeatV2Manager) upsertUnit(unit *client.Unit) {
 	cm.mx.Lock()
 	defer cm.mx.Unlock()
-	cm.units[unitKey{unit.Type(), unit.ID()}] = unit
+
+	aUnit, ok := cm.units[unitKey{unit.Type(), unit.ID()}]
+	if ok {
+		aUnit.update(unit)
+	} else {
+		aUnit = newAgentUnit(unit, cm.logger.Named("state-unit"))
```

Comment:
If you are going to create a logger for the unit, why not include the unit ID in the name?

Reply:
the inspiration came from the logger name of the `DiagCallback` but sure it makes sense to have the `ID` there, thus I added it. feel free to un-resolve if you spot anything